Realm/RLMSyncUser.mm

```diff
@@ -253,6 +254,7 @@ + (void)_performLogInForUser:(RLMSyncUser *)user
                     return;
                 }
                 user->_user = sync_user;
+                user.isAdmin = model.refreshToken.tokenData.isAdmin;
```

Comment:
Yes. There are actually two issues here:

- One is that, as you said, the flag should be tracked on the object store class instead. If the user fetches another `RLMSyncUser` object pointing to the same user, it won't have the proper flag set.
- The other is that the terminology is really confusing. We have the notion of a "client side admin user", which is a user who uses the admin token to open Realms, and a "server side admin user", which is a user flagged by ROS as an 'admin'. Perhaps we should rename the public API to "isObjectServerAdmin" or something similar.